File: scala/src/main/scala/tds/td2/modele/Annee.scala
package tds.td2.modele

// possible fuite de donnée + non fonctionnel
// modif modèle pour cohérence inscriptions
final case class Annee(ues: Set[UE]):

    var etudiants: Set[Etudiant] = Set()

    def inscrire(etudiant: Etudiant): Unit =
        etudiant.inscrire(this)
        etudiants = etudiants.union(Set(etudiant))

File: java/app/src/main/java/org/example/tds/td2/universite/Annee.java
package org.example.tds.td2.universite;

import java.util.Set;
import java.util.HashSet;

public class Annee {
    private Set<UE> ues;
    private Set<Etudiant> etudiants;

    public Annee(Set<UE> ues) {
        this.ues = new HashSet<>(ues); // éviter fuite données
        this.etudiants = new HashSet<>();
    }

    public Set<UE> ues() {
        return new HashSet<>(ues); // éviter fuite données
    }

    public Set<Etudiant> etudiants() {
        return new HashSet<>(etudiants); // éviter fuite données
    }

    public void inscrire(Etudiant e) {
        etudiants.add(e); // pas fonctionnel !
    }
}
